packages/core/src/parameter/scanner.ts

```diff
@@ -162,70 +162,133 @@ function asExternal(internal: InternalFlagName, scannerCaseStyle: ScannerCaseSty
     ) as ExternalFlagName;
 }
 
+export type ArgumentInputSource = "argv" | "default" | `env:${string}`;
+
+function redactInputFromProperties(obj: Record<string, unknown>, input: string, displayValue: string): void {
+    for (const propertyName of Object.getOwnPropertyNames(obj)) {
+        if (typeof obj[propertyName] === "string") {
+            obj[propertyName] = obj[propertyName].replaceAll(input, displayValue);
+        }
+    }
+}
+
 /**
  * Thrown when underlying parameter parser throws an exception parsing some input.
  */
 export class ArgumentParseError extends ArgumentScannerError {
+    static parseInput<T, P extends InputParser<T, CONTEXT>, CONTEXT extends CommandContext>(
```

Comment:
Slightly nit: I find these static methods on error classes a little odd as they are not always throwing errors.

Reply:
Yeah that's fair. I thought it would be nice to co-locate but now I'm also realize this means that the static function gets exported and it doesn't need to be part of the public API.